independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BuiltinBean.java

```diff
@@ -321,6 +327,19 @@ private static void generateInterceptedBeanBytecode(GeneratorContext ctx) {
                 interceptedBeanMetadataProviderSupplier);
     }
 
+    private static void generateDecoratedBeanBytecode(GeneratorContext ctx) {
+        ResultHandle decoratedBeanMetadataProvider = ctx.constructor
+                .newInstance(MethodDescriptor.ofConstructor(DecoratedBeanMetadataProvider.class));
+
+        ResultHandle decoratedBeanMetadataProviderSupplier = ctx.constructor.newInstance(
+                MethodDescriptors.FIXED_VALUE_SUPPLIER_CONSTRUCTOR, decoratedBeanMetadataProvider);
+        ctx.constructor.writeInstanceField(
+                FieldDescriptor.of(ctx.clazzCreator.getClassName(), ctx.providerName,
+                        Supplier.class.getName()),
+                ctx.constructor.getThis(),
+                decoratedBeanMetadataProviderSupplier);
+    }
+
```

Comment:
This method is basically a copy of `generateInterceptedBeanBytecode`, and the class `DecoratedBeanMetadataProvider` is basically a copy of `InterceptedBeanMetadataProvider`. I was wondering if we could merge them, but it's probably not a big deal if they stay as they are.